static/app/makeLazyloadComponent.tsx

```diff
@@ -17,19 +18,34 @@ export function makeLazyloadComponent<C extends React.ComponentType<any>>(
   resolve: () => Promise<{default: C}>,
   loadingFallback?: React.ReactNode
 ) {
-  const LazyComponent = lazy<C>(() => retryableImport(resolve));
+  // Create a shared promise that both lazy() and preload() will use
+  let sharedPromise: Promise<{default: C}> | null = null;
+  let loadedComponent: C | null = null;
+
+  const getSharedPromise = async () => {
+    if (!sharedPromise) {
+      sharedPromise = retryableImport(resolve);
+      loadedComponent = (await sharedPromise).default;
+    }
+    return sharedPromise;
+  };
+
+  const LazyComponent = lazy(getSharedPromise);
+
   // XXX: Assign the component to a variable so it has a displayname
   function RouteLazyLoad(props: React.ComponentProps<C>) {
-    // we can use this hook to set the organization as it's
-    // a child of the organization context
     return (
       <SafeLazyLoad
         {...props}
-        LazyComponent={LazyComponent}
+        // If the component is already loaded, render it directly to avoid Suspense
+        LazyComponent={loadedComponent ?? LazyComponent}
```

Comment:
This is great, and now I wonder if the tiny loader flickers on the already loaded routes were happening because we were always returning a promise here, meaning this would always resolve in the next micro task